test(smokehouse): add basic smoke test for SEO audits

brendankenny opened the pull request:
@kdzwinel FYI

this adds a starting point for SEO audit integration tests. Right now this tests only the two a11y audits that are also being used as SEO audits, but since they're a11y audits they don't have many details to test beyond the score

as new SEO audits get added:
- add audits to the expectations file (obviously :)
- new audits may have a lot more detail in `extendedInfo`/`details` that can be tested ([example](https://github.com/GoogleChrome/lighthouse/blob/a9b02a948b579be1a0667b69627f6d1f831d30ca/lighthouse-cli/test/smokehouse/pwa-expectations.js#L40-L56))
- could add a second html tester file that fails some/all of these audits and asserts proper failure output in those cases

brendankenny commented:
> @kdzwinel FYI

and @rviscomi et al :)

This also made me realize that our full-config will no longer be a config running all possible tests since the SEO config branches out separately, so we should probably figure that out at some point

patrickhulce commented:
I believe @rviscomi's plan was to nuke the SEO config and merge them into the full config when they were ready.

patrickhulce commented on package.json:
```diff
@@ -27,7 +27,7 @@
     "clean": "rimraf *.report.html *.report.dom.html *.report.json *.screenshots.html *.devtoolslog.json *.trace.json || true",
     "lint": "[ \"$CI\" = true ] && eslint --quiet -f codeframe . || eslint .",
-    "smoke": "bash lighthouse-cli/test/smokehouse/offline-local/run-tests.sh && bash lighthouse-cli/test/smokehouse/perf/run-tests.sh && bash lighthouse-cli/test/smokehouse/dobetterweb/run-tests.sh && bash lighthouse-cli/test/smokehouse/byte-efficiency/run-tests.sh && bash lighthouse-cli/test/smokehouse/tricky-ttci/run-tests.sh",
+    "smoke": "bash lighthouse-cli/test/smokehouse/offline-local/run-tests.sh && bash lighthouse-cli/test/smokehouse/perf/run-tests.sh && bash lighthouse-cli/test/smokehouse/dobetterweb/run-tests.sh && bash lighthouse-cli/test/smokehouse/byte-efficiency/run-tests.sh && bash lighthouse-cli/test/smokehouse/tricky-ttci/run-tests.sh && bash lighthouse-cli/test/smokehouse/seo/run-tests.sh",
```
we're gonna need a bash script to run our bash scripts 😆

paulirish replied:
I looked at shortening this to...

```sh
for file in lighthouse-cli/test/smokehouse/*/run-tests.sh; do bash $file; done
``` 

however it doesnt' handle the exit code correctly, so... we'll just need to make run-test.js not terrible sometime. perhaps a job for lighthouse-assert..

patrickhulce commented on lighthouse-cli/test/fixtures/seo/seo-tester.html:
```diff
@@ -0,0 +1,28 @@
+<!doctype html>
+<!--
+ * Copyright 2017 Google Inc. All rights reserved.
```
small license?

brendankenny replied:
> small license?

our HTML files don't appear to have a version of the small license and I didn't care enough to make one. Good first bug! :)

paulirish replied:
there u go: https://github.com/GoogleChrome/lighthouse/pull/3309/files

brendankenny replied:
😶

patrickhulce commented on lighthouse-cli/test/smokehouse/seo/expectations.js:
```diff
@@ -0,0 +1,24 @@
+    url: 'http://localhost:10200/seo/seo-tester.html',
+    audits: {
+      'meta-viewport': {
+        score: true,
```
do we want the smoke test to be catching failures or passes?

brendankenny replied:
> do we want the smoke test to be catching failures or passes?

yeah, I was just dropping this in to get things going, but I can add in a failure page too

brendankenny commented:
added
- test for just landed meta-description audit
- a second page to check SEO audit failures

In the `seo.js` config, switched the `meta-viewport` audit (an a11y audit) with `viewport` audit, which checks for valid viewport content @rviscomi @kdzwinel

brendankenny commented:
failing tests are just the JVM issue facing all builds right now...
